services/w3c/w3c-validation-helper.js

```diff
@@ -0,0 +1,157 @@
+  const schema = []
+  const decodedPreset = decodeURI(preset)
+  if (new RegExp(html4Expression, 'i').test(decodedPreset)) {
+    if (/Strict/i.test(decodedPreset)) {
+      schema.push('http://s.validator.nu/xhtml10/xhtml-strict.rnc')
+    } else if (/Transitional/i.test(decodedPreset)) {
+      schema.push('http://s.validator.nu/xhtml10/xhtml-transitional.rnc')
+    } else {
+      schema.push('http://s.validator.nu/xhtml10/xhtml-frameset.rnc')
+    }
+    schema.push('http://s.validator.nu/html4/assertions.sch')
+    schema.push('http://c.validator.nu/all-html4/')
+  } else if (/1\.0 Strict, URL, Ruby, SVG 1\.1/i.test(decodedPreset)) {
+    schema.push('http://s.validator.nu/xhtml1-ruby-rdf-svg-mathml.rnc')
+    schema.push('http://s.validator.nu/html4/assertions.sch')
+    schema.push('http://c.validator.nu/all-html4/')
+  } else {
+    if (new RegExp(html5Expression, 'i').test(decodedPreset)) {
+      if (/ITS 2\.0/i.test(decodedPreset)) {
+        schema.push('http://s.validator.nu/html5-its.rnc')
+      } else if (/RDFa Lite 1\.1/i.test(decodedPreset)) {
+        schema.push('http://s.validator.nu/html5-rdfalite.rnc')
+      } else {
+        schema.push('http://s.validator.nu/html5.rnc')
+      }
+    } else if (new RegExp(xhtmlExpression, 'i').test(decodedPreset)) {
+      if (/RDFa Lite 1\.1/i.test(decodedPreset)) {
+        schema.push('http://s.validator.nu/xhtml5-rdfalite.rnc')
+      } else {
+        schema.push('http://s.validator.nu/xhtml5.rnc')
+      }
+    } else if (new RegExp(svgExpression, 'i').test(decodedPreset)) {
+      schema.push('http://s.validator.nu/svg-xhtml5-rdf-mathml.rnc')
+    }
+    schema.push('http://s.validator.nu/html5/assertions.sch')
+    schema.push('http://c.validator.nu/all/')
+  }
+  return schema.join(' ')
+}
+
+const documentation = `
+  <style>
+    .box {
+      display: flex;
+      justify-content: space-between;
+    }
+    .note {
+      font-size: smaller;
+      text-align: left;
+    }
+  </style>
+  <p>
+    The W3C validation badge performs validation of the HTML, SVG, MathML, ITS, RDFa Lite, XHTML documents.
+    The badge uses the type property of each message found in the messages from the validation results to determine to be an error or warning.
+    The rules are as follows:
+    <ul class="note">
+      <li>info:  These messages are counted as warnings</li>
+      <li>error:  These messages are counted as errors</li>
+      <li>non-document-error: These messages are counted as errors</li>
+    </ul>
+  </p>
+  <p>
+    This badge relies on the https://validator.nu service to perform the validation. Please refer to https://about.validator.nu/ for the full documentation and Terms of service.
+    The following are required from the consumer for the badge to function.
+
+    <ul class="note">
+      <li>
+        Path:
+        <ul>  
+          <li>
+            parser (Optional can be left as blank): The parser that is used for validation. This is a passthru value to the service
```

Comment:
This text should probably be updated or removed now correct? Perhaps we could mention how `default` works, but with the current badge route definition one of those four values must be specified

Reply:
Updated the text. I left it in for better explanation. I can remove it if required

Reply:
Thanks, I like this as is. It provides helpful context to folks, especially the explanation of `default` 👍